File: src/main/java/com/example/venka/javaspringsparkdemo/dao/MessageDAO.java
package com.example.venka.javaspringsparkdemo.dao;

import com.example.venka.javaspringsparkdemo.model.Message;
import org.bson.types.ObjectId;

import java.util.List;

public interface MessageDAO {

    void insert(final Message message);

    void delete(final ObjectId id);

    List<Message> getAll();

    Message getById(final ObjectId id);
}

File: src/main/kotlin/com/example/venka/kotlinspringsparkdemo/dao/MessageDAO.kt
package com.example.venka.kotlinspringsparkdemo.dao

import com.example.venka.kotlinspringsparkdemo.model.Message
import org.bson.types.ObjectId

interface MessageDAO {

    val all: List<Message>

    fun insert(message: Message)

    fun delete(id: ObjectId)

    fun getById(id: ObjectId): Message
}
